src/common/session/session-entity.js

```diff
@@ -42,31 +42,37 @@ export class SessionEntity {
    * expiresMs and inactiveMs are used to "expire" the session, but can be overridden in the constructor. Pass 0 to disable expiration timers.
    */
   constructor (opts) {
-    this.setup(opts)
-  }
+    const { agentIdentifier, key, storage } = opts
 
-  setup ({ agentIdentifier, key, storage, value = generateRandomHexString(16), expiresMs = DEFAULT_EXPIRES_MS, inactiveMs = DEFAULT_INACTIVE_MS }) {
     if (!agentIdentifier || !key || !storage) {
       throw new Error(`Missing required field(s):${!agentIdentifier ? ' agentID' : ''}${!key ? ' key' : ''}${!storage ? ' storage' : ''}`)
     }
     this.agentIdentifier = agentIdentifier
     this.storage = storage
     this.state = {}
 
-    this.sync(model)
-
     // key is intended to act as the k=v pair
     this.key = key
+
+    // Only setup EE and wrap events once per instance
+    this.ee = ee.get(agentIdentifier)
+    wrapEvents(this.ee)
+
+    this.setup(opts)
+  }
+
+  setup ({ value = generateRandomHexString(16), expiresMs = DEFAULT_EXPIRES_MS, inactiveMs = DEFAULT_INACTIVE_MS }) {
+    console.log('Session.setup')
```

Comment:
Console

Reply:
Shit I thought I got all of them lol thanks